src/common/io.py

```diff
@@ -1477,6 +1475,125 @@ def get_variable_data(self, name):
+                    then the behavior is the default Python slice-behavior.
+
+            Returns::
+                Tuple: (List of trajectories, next start index (non-negative))
+        """
+
+        """
+            TODO:
+            For [start_index, stop_index] we can do performance improvements
+            since methods such as _get_trajectory, get_diagnostics_trajectory
+            and _get_interpolated_trajectory always retrieve the full
+            trajectory which is then sliced, here. Instead we can account for
+            start_index and stop_index and reduce the time spent reading unused data points.
+        """
+        trajectories = []
+
+        # Get the time trajectory
+        if not self._contains_diagnostic_data:
+            tt = self._get_trajectory(0)[start_index:stop_index]
+        else:
+            # Since we interpolate data if diagnostics is enabled
+            tt = self._get_diagnostics_trajectory(0)[start_index:stop_index]
+
+        # Need to account for data that might be added while we are iterating over 'names' later
+        if stop_index is None:
+            stop_index = len(tt) + start_index
+
+        for name in names:
+            if isinstance(name, bytes):
+                name = decode(name)
+
+            if name == 'time' or name == 'Time':
+                trajectories.append(Trajectory(tt, tt))
+                continue
+            elif self._contains_diagnostic_data and (
+                    name in [
+                        DiagnosticsBase.calculated_diagnostics['nbr_events']['name'],
+                        DiagnosticsBase.calculated_diagnostics['nbr_time_events']['name'],
+                        DiagnosticsBase.calculated_diagnostics['nbr_state_events']['name'],
+                        DiagnosticsBase.calculated_diagnostics['nbr_steps']['name']]
+                ):
+                trajectories.append(
+                    Trajectory(tt, self._calculate_events_and_steps(name)[start_index:stop_index])
+                )
+                continue
+            elif self._contains_diagnostic_data and (
+                    f"{DiagnosticsBase.calculated_diagnostics['nbr_state_limits_step']['name']}." in name
+                ):
+                trajectories.append(
+                    Trajectory(tt, self._calculate_nbr_state_limits_step(name)[start_index:stop_index])
+                )
+                continue
+            elif self._contains_diagnostic_data and (
+                    name == f'{DIAGNOSTICS_PREFIX}cpu_time'
+                ):
+                trajectories.append(
+                    Trajectory(
+                        tt, self.get_variable_data(f'{DIAGNOSTICS_PREFIX}cpu_time_per_step').x[start_index:stop_index]
+                    )
+                )
+                continue
+            else:
+                # expect potential issues with diagnostics data right here
```

Comment:
I am removing this line `# expect potential issues...` since it is redundant now.